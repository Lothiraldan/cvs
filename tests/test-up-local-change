#!/bin/sh

export HGMERGE=true

set -ex
mkdir r1
cd r1
hg init
echo a > a
hg addremove
hg commit -t "1" -u test -d "0 0"

hg clone . ../r2
cd ../r2
hg up
echo abc > a
hg diff > ../d
sed "s/\(\(---\|+++\).*\)\t.*/\1/" < ../d

cd ../r1
echo b > b
echo a2 > a
hg addremove
hg commit -t "2" -u test -d "0 0"

cd ../r2
hg -q pull ../r1
hg status
hg -d up
hg -d up -m
hg parents
hg -v history
hg diff > ../d
sed "s/\(\(---\|+++\).*\)\t.*/\1/" < ../d

