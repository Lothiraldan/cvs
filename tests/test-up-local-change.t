  $ HGMERGE=true; export HGMERGE

  $ hg init r1
  $ cd r1
  $ echo a > a
  $ hg addremove
  adding a
  $ hg commit -m "1"

  $ hg clone . ../r2
  updating to branch default
  1 files updated, 0 files merged, 0 files removed, 0 files unresolved
  $ cd ../r2
  $ hg up
  0 files updated, 0 files merged, 0 files removed, 0 files unresolved
  $ echo abc > a
  $ hg diff --nodates
  diff -r c19d34741b0a a
  --- a/a
  +++ b/a
  @@ -1,1 +1,1 @@
  -a
  +abc

  $ cd ../r1
  $ echo b > b
  $ echo a2 > a
  $ hg addremove
  adding b
  $ hg commit -m "2"

  $ cd ../r2
  $ hg -q pull ../r1
  $ hg status
  M a
  $ hg parents
  changeset:   0:c19d34741b0a
  user:        test
  date:        Thu Jan 01 00:00:00 1970 +0000
  summary:     1
  
  $ hg --debug up
    searching for copies back to rev 1
    unmatched files in other:
     b
  resolving manifests
   branchmerge: False, force: False, partial: False
   ancestor: c19d34741b0a, local: c19d34741b0a+, remote: 1e71731e6fbb
   preserving a for resolve of a
   b: remote created -> g
  getting b
  updating: b 1/2 files (50.00%)
   a: versions differ -> m
  updating: a 2/2 files (100.00%)
  picked tool 'true' for a (binary False symlink False)
  merging a
  my a@c19d34741b0a+ other a@1e71731e6fbb ancestor a@c19d34741b0a
  launching merge tool: true $TESTTMP/r2/a * (glob)
  merge tool returned: 0
  1 files updated, 1 files merged, 0 files removed, 0 files unresolved
  $ hg parents
  changeset:   1:1e71731e6fbb
  tag:         tip
  user:        test
  date:        Thu Jan 01 00:00:00 1970 +0000
  summary:     2
  
  $ hg --debug up 0
  resolving manifests
   branchmerge: False, force: False, partial: False
   ancestor: 1e71731e6fbb, local: 1e71731e6fbb+, remote: c19d34741b0a
   preserving a for resolve of a
   b: other deleted -> r
  removing b
  updating: b 1/2 files (50.00%)
   a: versions differ -> m
  updating: a 2/2 files (100.00%)
  picked tool 'true' for a (binary False symlink False)
  merging a
  my a@1e71731e6fbb+ other a@c19d34741b0a ancestor a@1e71731e6fbb
  launching merge tool: true $TESTTMP/r2/a * (glob)
  merge tool returned: 0
  0 files updated, 1 files merged, 1 files removed, 0 files unresolved
  $ hg parents
  changeset:   0:c19d34741b0a
  user:        test
  date:        Thu Jan 01 00:00:00 1970 +0000
  summary:     1
  
  $ hg parents
  changeset:   0:c19d34741b0a
  user:        test
  date:        Thu Jan 01 00:00:00 1970 +0000
  summary:     1
  
  $ hg --debug up
    searching for copies back to rev 1
    unmatched files in other:
     b
  resolving manifests
   branchmerge: False, force: False, partial: False
   ancestor: c19d34741b0a, local: c19d34741b0a+, remote: 1e71731e6fbb
   preserving a for resolve of a
   b: remote created -> g
  getting b
  updating: b 1/2 files (50.00%)
   a: versions differ -> m
  updating: a 2/2 files (100.00%)
  picked tool 'true' for a (binary False symlink False)
  merging a
  my a@c19d34741b0a+ other a@1e71731e6fbb ancestor a@c19d34741b0a
  launching merge tool: true $TESTTMP/r2/a * (glob)
  merge tool returned: 0
  1 files updated, 1 files merged, 0 files removed, 0 files unresolved
  $ hg parents
  changeset:   1:1e71731e6fbb
  tag:         tip
  user:        test
  date:        Thu Jan 01 00:00:00 1970 +0000
  summary:     2
  
  $ hg -v history
  changeset:   1:1e71731e6fbb
  tag:         tip
  user:        test
  date:        Thu Jan 01 00:00:00 1970 +0000
  files:       a b
  description:
  2
  
  
  changeset:   0:c19d34741b0a
  user:        test
  date:        Thu Jan 01 00:00:00 1970 +0000
  files:       a
  description:
  1
  
  
  $ hg diff --nodates
  diff -r 1e71731e6fbb a
  --- a/a
  +++ b/a
  @@ -1,1 +1,1 @@
  -a2
  +abc


create a second head

  $ cd ../r1
  $ hg up 0
  1 files updated, 0 files merged, 1 files removed, 0 files unresolved
  $ echo b2 > b
  $ echo a3 > a
  $ hg addremove
  adding b
  $ hg commit -m "3"
  created new head

  $ cd ../r2
  $ hg -q pull ../r1
  $ hg status
  M a
  $ hg parents
  changeset:   1:1e71731e6fbb
  user:        test
  date:        Thu Jan 01 00:00:00 1970 +0000
  summary:     2
  
  $ hg --debug up
  abort: uncommitted changes
  (commit and merge, or update --clean to discard changes)
  [255]

test conflicting untracked files

  $ hg up -qC 0
  $ echo untracked > b
  $ hg st
  ? b
  $ hg up 1
  b: untracked file differs
  abort: untracked files in working directory differ from files in requested revision
  [255]
  $ rm b

test conflicting untracked ignored file

  $ hg up -qC 0
  $ echo ignored > .hgignore
  $ hg add .hgignore
  $ hg ci -m 'add .hgignore'
  created new head
  $ echo ignored > ignored
  $ hg add ignored
  $ hg ci -m 'add ignored file'

  $ hg up -q 'desc("add .hgignore")'
  $ echo untracked > ignored
  $ hg st
  $ hg up 'desc("add ignored file")'
  ignored: untracked file differs
  abort: untracked files in working directory differ from files in requested revision
  [255]

test a local add

  $ cd ..
  $ hg init a
  $ hg init b
  $ echo a > a/a
  $ echo a > b/a
  $ hg --cwd a commit -A -m a
  adding a
  $ cd b
  $ hg add a
  $ hg pull -u ../a
  pulling from ../a
  requesting all changes
  adding changesets
  adding manifests
  adding file changes
  added 1 changesets with 1 changes to 1 files
  1 files updated, 0 files merged, 0 files removed, 0 files unresolved
  $ hg st

  $ cd ..
