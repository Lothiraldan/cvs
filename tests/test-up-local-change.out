adding a
1 files updated, 0 files merged, 0 files removed, 0 files unresolved
0 files updated, 0 files merged, 0 files removed, 0 files unresolved
diff -r 33aaa84a386b a
--- a/a
+++ b/a
@@ -1,1 +1,1 @@ a
-a
+abc
adding b
M a
changeset:   0:33aaa84a386b
user:        test
date:        Mon Jan 12 13:46:40 1970 +0000
summary:     1

resolving manifests
 force None allow None moddirstate True linear True
 ancestor a0c8bcbbb45c local a0c8bcbbb45c remote 1165e8bd193e
 a versions differ, resolve
remote created b
getting b
merging a
resolving a
file a: my b789fdd96dc2 other d730145abbf9 ancestor b789fdd96dc2
1 files updated, 1 files merged, 0 files removed, 0 files unresolved
changeset:   1:802f095af299
tag:         tip
user:        test
date:        Mon Jan 12 13:46:40 1970 +0000
summary:     2

resolving manifests
 force None allow None moddirstate True linear True
 ancestor a0c8bcbbb45c local 1165e8bd193e remote a0c8bcbbb45c
remote deleted b
removing b
0 files updated, 0 files merged, 1 files removed, 0 files unresolved
changeset:   0:33aaa84a386b
user:        test
date:        Mon Jan 12 13:46:40 1970 +0000
summary:     1

abort: there is nothing to merge, just use 'hg update'
failed
changeset:   0:33aaa84a386b
user:        test
date:        Mon Jan 12 13:46:40 1970 +0000
summary:     1

resolving manifests
 force None allow None moddirstate True linear True
 ancestor a0c8bcbbb45c local a0c8bcbbb45c remote 1165e8bd193e
 a versions differ, resolve
remote created b
getting b
merging a
resolving a
file a: my b789fdd96dc2 other d730145abbf9 ancestor b789fdd96dc2
1 files updated, 1 files merged, 0 files removed, 0 files unresolved
changeset:   1:802f095af299
tag:         tip
user:        test
date:        Mon Jan 12 13:46:40 1970 +0000
summary:     2

changeset:   1:802f095af299cde27a85b2f056aef3829870956c
tag:         tip
user:        test
date:        Mon Jan 12 13:46:40 1970 +0000
files:       a b
description:
2


changeset:   0:33aaa84a386bd609094aeb21a97c09436c482ef1
user:        test
date:        Mon Jan 12 13:46:40 1970 +0000
files:       a
description:
1


diff -r 802f095af299 a
--- a/a
+++ b/a
@@ -1,1 +1,1 @@ a2
-a2
+abc
1 files updated, 0 files merged, 1 files removed, 0 files unresolved
adding b
M a
changeset:   1:802f095af299
user:        test
date:        Mon Jan 12 13:46:40 1970 +0000
summary:     2

resolving manifests
 force None allow None moddirstate True linear False
 ancestor a0c8bcbbb45c local 1165e8bd193e remote 4096f2872392
 a versions differ, resolve
 b versions differ, resolve
this update spans a branch affecting the following files:
 a (resolve)
 b (resolve)
aborting update spanning branches!
(use 'hg merge' to merge across branches or 'hg update -C' to lose changes)
failed
abort: outstanding uncommitted changes
failed
resolving manifests
 force None allow 1 moddirstate True linear False
 ancestor a0c8bcbbb45c local 1165e8bd193e remote 4096f2872392
 a versions differ, resolve
 b versions differ, resolve
merging a
resolving a
file a: my d730145abbf9 other 13e0d5f949fa ancestor b789fdd96dc2
merging b
resolving b
file b: my 1e88685f5dde other 61de8c7723ca ancestor 000000000000
0 files updated, 2 files merged, 0 files removed, 0 files unresolved
(branch merge, don't forget to commit)
changeset:   1:802f095af299
user:        test
date:        Mon Jan 12 13:46:40 1970 +0000
summary:     2

changeset:   2:030602aee63d
tag:         tip
parent:      0:33aaa84a386b
user:        test
date:        Mon Jan 12 13:46:40 1970 +0000
summary:     3

diff -r 802f095af299 a
--- a/a
+++ b/a
@@ -1,1 +1,1 @@ a2
-a2
+abc
adding a
pulling from ../a
requesting all changes
adding changesets
adding manifests
adding file changes
added 1 changesets with 1 changes to 1 files
1 files updated, 0 files merged, 0 files removed, 0 files unresolved
